plugins/catalog-backend-module-ldap/src/ldap/config.ts

```diff
@@ -208,6 +208,10 @@ export function readLdapConfig(config: Config): LdapProviderConfig[] {
       scope: c.getOptionalString('scope') as SearchOptions['scope'],
       filter: formatFilter(c.getOptionalString('filter')),
       attributes: c.getOptionalStringArray('attributes'),
+      sizeLimit: c.getOptionalNumber('sizeLimit'),
+      timeLimit: c.getOptionalNumber('timeLimit'),
+      derefAliases: c.getOptionalNumber('derefAliases'),
+      typesOnly: c.getOptionalBoolean('typesOnly'),
```

Comment:
Not seeing these in the config schema, let's add them there too. Is there a reason weren't there to begin with @freben btw?

Reply:
Argh - missed that. Added them 👍

Reply:
From my point of view those options should rarely be used - in our case we currently only need the `timeLimit` option, but I added all as the used `SearchOptions` type also supports all of them.

Reply:
I think the original thinking was that I wanted to go a bit carefully because this turns out to be 1:1 with a third party lib whose API will move over time and break this contract in ways that are out of our control. Once we start supporting a parameter externally we are also responsible for maintaining it. So smaller surface does have some benefits, so we don't sit there and have to make breaking changes in unrelated places just because we needed to make a big security bump of the lib or so :)

That being said I'm not really saying no either. If this is useful for real, then let's go.

Reply:
Yeah, understandable 👍 So far I see it, the options are not library specific (see [LDAP options](https://ldapwiki.com/wiki/SearchRequest)), they are directly [handed over to LDAP](https://github.com/ldapjs/node-ldapjs/blob/91172e3929e25a53398a66fd1774da88aaff4a6a/lib/messages/search_request.js#L118) itself.

Nevertheless, I would be fine with only adding `timeLimit` - we need at least that one, as we currently have some LDAP queries which take ~16s 😬

Reply:
Nah this looks good. It's a bit annoying that there's no helpful typing on the interface. For example, if we wrote this ourselves we would have made the timeLimit be a Duration after parsing. Just seeing timeLimit doesn't tell you if it's milliseconds or seconds or something else. But since we do absolutely nothing but pass it through, it's probably just as well to leave them alone and lean on people perhaps being familiar with the ldap library parameters and appreciating that they're the same. We do that for the connection block to knex database settings too for example.

So ... yeah let's leave it as is